src/app/(proper_react)/(redesign)/(authenticated)/admin/churn-subscribers/actions.tsx

```diff
@@ -9,17 +9,34 @@ import {
   deleteSubscriberChurns,
 } from "../../../../../../db/tables/subscriber_churns";
 import { SubscriberChurnRow } from "knex/types/tables";
+import { getServerSession } from "../../../../../functions/server/getServerSession";
+import { isAdmin } from "../../../../../api/utils/auth";
 
 export async function getAllChurns() {
+  const session = await getServerSession();
+  if (!session?.user?.email || !isAdmin(session.user.email)) {
+    return [];
+  }
+
   return getAllSubscriberChurns();
 }
 
 export async function upsertAllChurns(
   churningSubscribers: SubscriberChurnRow[],
 ) {
+  const session = await getServerSession();
+  if (!session?.user?.email || !isAdmin(session.user.email)) {
```

Comment:
Is there an advantage to `(!session?.user?.email || !isAdmin(session.user.email))` vs. just `(!isAdmin(session?.user?.email)`?

Reply:
moved them to a separate func